apis/apps/v1alpha1/cluster_types.go

```diff
@@ -162,11 +163,6 @@ type ClusterComponentSpec struct {
 	// +optional
 	Services []ClusterComponentService `json:"services,omitempty"`
 
-	// primaryIndex determines which index is primary when workloadType is Replication. Index number starts from zero.
-	// +kubebuilder:validation:Minimum=0
-	// +optional
-	PrimaryIndex *int32 `json:"primaryIndex,omitempty"`
-
 	// switchPolicy defines the strategy for switchover and failover when workloadType is Replication.
```

Comment:
is this still `Replication` workload only attribute?

Reply:
> is this still `Replication` workload only attribute?

According to the current design, yes.